iroh-net/src/magicsock/node_map/node_state.rs

```diff
@@ -1227,64 +1271,85 @@ impl From<RelayUrlInfo> for RelayUrl {
     }
 }
 
-/// Details about an iroh node which is known to this node.
+/// Details about a remote iroh-net node which is known to this node.
+///
+/// Having details of a node does not mean it can be connected to.  Nor that it has ever
+/// been connected to in the past.  There are various reasons a node might be known: it
+/// could have been manually added via [`Endpoint::add_node_addr`], it could have been added
+/// by some discovery mechanism.
+///
+/// [`Endpoint::add_node_addr`]: crate::endpoint::Endpoint::add_node_addr
 #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
 pub struct NodeInfo {
-    /// The id in the node_map
-    pub id: usize,
-    /// The public key of the endpoint.
+    /// The globally unique public identifier for this node.
     pub node_id: NodeId,
-    /// relay server information, if available.
+    /// Relay server information, if available.
     pub relay_url: Option<RelayUrlInfo>,
-    /// List of addresses at which this node might be reachable, plus any latency information we
-    /// have about that address and the last time the address was used.
+    /// The addresses at which this node might be reachable.
+    ///
+    /// Some of these addresses might only be valid for networks we are not part of, but the
+    /// remote node might be.
     pub addrs: Vec<DirectAddrInfo>,
     /// The type of connection we have to the node, either direct or over relay.
     pub conn_type: ConnectionType,
-    /// The latency of the `conn_type`.
+    /// The latency of the current network path to the remote node.
     pub latency: Option<Duration>,
-    /// Duration since the last time this node was used.
+    /// Time elapsed time since last sending to, or received from the node.
+    ///
+    /// This is the duration since *any* data was sent or receive from the remote node,
+    /// payload or control messages.  Note that sending to the remote node does not imply
+    /// the remote node received anything.
     pub last_used: Option<Duration>,
 }
 
 impl NodeInfo {
     /// Get the duration since the last activity we received from this endpoint
     /// on any of its direct addresses.
+    // TODO: This does not contain what it claims to contain!  It can not be fixed easily.
+    #[deprecated(since = "0.23.0", note = "this is broken")]
```

Comment:
I am really against this, we shouldn't deprecate methods without having replacements, just because we think/know they are broken

Reply:
reverted in e6cb86313862fbb9cf9e9c6cb096436f9b3e3163